### PR TITLE
refactor: reduce dependencies and remove metabolite ID suffixes

### PR DESCRIPTION
**Pull request by edkerk**
### Main improvements in this PR:
To reduce dependencies, the model I/O is now completely running on RAVEN, as it previously dependent both on RAVEN and COBRA Toolboxes. All the functions in `code/modelTests` and `code/otherChanges` also run solely via RAVEN. `loadYeastModel` now has a `cobra`-flag that gives the model in COBRA format instead (default is RAVEN). Not all other code is COBRA-free yet, but this will be done in later PRs when required.

Requires [RAVEN 2.6.2](https://github.com/SysBioChalmers/RAVEN/releases/tag/v2.6.2).

In addition, the compartment information was removed from the metabolite names and metabolite IDs. This was a remnant of early COBRA-style models, but compartment information is now properly stored elsewhere. Meanwhile, this previous behaviour could cause conflicts, because even if a metabolite's compartment is changed in `model.metComps`, it would still have the previous compartment information in `model.metNames` and `model.mets`. While this will break some backwards compatibility with older code (which can easily be solved on a case-by-case basis), it would be good to get rid of this old baggage.

**I hereby confirm that I have:**

- [x] Tested my code with [all requirements](https://github.com/SysBioChalmers/yeast-GEM#required-software---contributor) for running the model
- [x] Selected `develop` as a target branch (top left drop-down menu)
- [ ] If needed, asked first in the [Gitter chat room](https://gitter.im/SysBioChalmers/yeast-GEM) about this PR

### CODE REVIEW

File: code/modelTests/essentialGenes.m [hunk 1]
```diff
@@ -15,6 +15,7 @@
 initCobraToolbox
```
Comment (feiranl) [inline code/modelTests/essentialGenes.m:15]: cobra function `initCobraToolbox `, which will be fixed in a separate PR

Comment (feiranl) [inline code/modelTests/essentialGenes.m:15]: cobra function `initCobraToolbox `, which will be fixed in a separate PR

File: code/modelTests/findDuplicatedRxns.m [hunk 2]
```diff
@@ -9,14 +9,13 @@ function findDuplicatedRxns(model)
 %   Usage: findDuplicatedRxns(model)
 %
 
-model_r = ravenCobraWrapper(model);
 for i = 1:length(model.rxns)-1
     for j = i+1:length(model.rxns)
         if isequal(model.S(:,i),model.S(:,j)) || isequal(model.S(:,i),-model.S(:,j))
             printRxnFormula(model,model.rxns(i),true,true,true);
```
Comment (feiranl) [inline code/modelTests/findDuplicatedRxns.m:15]: cobra function `printRxnFormula` , which will be fixed in a separate PR

**Comment by hongzhonglu** [comment]
Just go through the update, looks great and make life easier!  As "I/O is now completely running on RAVEN",  it shows COBRA could replace RAVEN if the user did not install RAVEN. I guess the model format is still consistent with cobrapy?

**Comment by edkerk** [comment]
Indeed, this is exactly why sorting was introduced in https://github.com/SysBioChalmers/RAVEN/pull/364, and this will never occur again. This is the first time that sorting is applied to the model, so having large changes now is unavoidable.

I do realize now that sorting the file and removing the metabolic suffices could have been separate PRs to have made things a _little_ more clear.

**Comment by mihai-sysbio** [comment]
> this will never occur again

awesome!
